src/main/java/core/basesyntax/dao/machine/MachineDaoImpl.java

```diff
@@ -12,11 +15,36 @@ public MachineDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Machine save(Machine machine) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.save(machine);
+            transaction.commit();
+            return machine;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Could not save machine "
+                    + machine + " item. ", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
 
     @Override
     public List<Machine> findByAgeOlderThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            return session.createQuery("from Machine where year < :age", Machine.class)
```

Comment:
```suggestion
            return session.createQuery("from Machine where year < :dateFrom", Machine.class)
```
I think it is more understandable